internal/stanza/converter.go

```diff
@@ -246,56 +229,44 @@ func (c *Converter) workerLoop() {
 func (c *Converter) batchLoop() {
 	defer c.wg.Done()
 
-	ticker := time.NewTicker(c.flushInterval)
-	defer ticker.Stop()
-
 	for {
 		select {
-		case wi, ok := <-c.batchChan:
+		case workerItems, ok := <-c.batchChan:
 			if !ok {
 				return
 			}
 
-			pLogs, ok := c.data[wi.ResourceID]
-			if ok {
-				lr := pLogs.ResourceLogs().
-					At(0).InstrumentationLibraryLogs().
-					At(0).Logs().AppendEmpty()
-				wi.LogRecord.CopyTo(lr)
-			} else {
-				pLogs = pdata.NewLogs()
-				logs := pLogs.ResourceLogs()
-				rls := logs.AppendEmpty()
-
-				resource := rls.Resource()
-				resourceAtts := resource.Attributes()
-				resourceAtts.EnsureCapacity(len(wi.Resource))
-				for k, v := range wi.Resource {
-					resourceAtts.InsertString(k, v)
+			for _, wi := range workerItems {
+				pLogs, ok := c.data[wi.ResourceID]
+				if ok {
+					lr := pLogs.ResourceLogs().
+						At(0).InstrumentationLibraryLogs().
+						At(0).Logs().AppendEmpty()
+					wi.LogRecord.CopyTo(lr)
+				} else {
+					pLogs = pdata.NewLogs()
+					logs := pLogs.ResourceLogs()
+					rls := logs.AppendEmpty()
+
+					resource := rls.Resource()
+					resourceAtts := resource.Attributes()
+					resourceAtts.EnsureCapacity(len(wi.Resource))
+					for k, v := range wi.Resource {
+						resourceAtts.InsertString(k, v)
+					}
+
+					ills := rls.InstrumentationLibraryLogs()
+					lr := ills.AppendEmpty().Logs().AppendEmpty()
+					wi.LogRecord.CopyTo(lr)
 				}
```

Comment:
Continue early to skip the else and keep the code flat
```suggestion
				pLogs, ok := c.data[wi.ResourceID]
				if ok {
					lr := pLogs.ResourceLogs().
						At(0).InstrumentationLibraryLogs().
						At(0).Logs().AppendEmpty()
					wi.LogRecord.CopyTo(lr)
					continue
				}

				pLogs = pdata.NewLogs()
				logs := pLogs.ResourceLogs()
				rls := logs.AppendEmpty()

				resource := rls.Resource()
				resourceAtts := resource.Attributes()
				resourceAtts.EnsureCapacity(len(wi.Resource))
				for k, v := range wi.Resource {
					resourceAtts.InsertString(k, v)
				}

				ills := rls.InstrumentationLibraryLogs()
				lr := ills.AppendEmpty().Logs().AppendEmpty()
				wi.LogRecord.CopyTo(lr)
```